### DISCUSSION TITLE
JupyterBook - glue:figure problem with DataFrame

### DISCUSSION THREAD

**Discussion by luisafuss** [executablebooks/meta#207] [Q&A]
Hey everyone,
I'm trying to toggle a DataFrame in a JupyterBook. However, I run into a few problems with this.
That's how the DataFrame looks:
![image](https://user-images.githubusercontent.com/54306427/101925296-fdf7a100-3bd1-11eb-9d65-ed4bd86d8e32.png)

And if I try 
```
glue("df", df, display=False)
```

and then
```
{glue:}`df`
```

It just looks like this:
![image](https://user-images.githubusercontent.com/54306427/101925770-88d89b80-3bd2-11eb-9c45-e608e312ebf0.png)

When I instead try to use 
```
{glue:figure}`df`
```

It says "WARNING: Unknown interpreted text role "glue:figure"."

I've installed the latest version of myst-nb.

Do you have an idea of what the problem could be?
Thanks in advance!

**Comment by choldgraf** [comment]
Hmmm that’s weird - could there be an issue because of the html in some of th cells? If you remove that HTML so it’s just markdown in each cell, does it still not work?

**Reply by luisafuss** [reply]
No, I think that's not the problem. 
I tried it with a minimal example and the issue is the same.

![image](https://user-images.githubusercontent.com/54306427/102072654-793f9980-3e02-11eb-83e3-7050b45dfd77.png)

Command used: {glue:}`df2`

**Reply by choldgraf** [reply]
hmmm - in that case this is feeling like a bug to me (or at least, something that we need to document better). Could you open up an issue in [the myst-nb repository](https://github.com/executablebooks/MyST-NB) so that we can track it?

**Reply by luisafuss** [reply]
Okay, thank you!

**Comment by luisafuss** [comment]
I solved the issue with this tip: https://github.com/executablebooks/MyST-NB/issues/282
